### PR TITLE
Rename API for changing BackHandler behavior

### PR DESCRIPTION
**Pull request by acoates-ms**
Feedback from @asklar on #8365.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8385)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/QuirkSettings.idl [hunk 1]
```diff
@@ -28,10 +34,10 @@ namespace Microsoft.ReactNative
...
 
     DOC_STRING(
-      "Runtime setting that can be used to skip the default back handler logic.  "
-      "If you have an application that wants to handle back bahavior in native code "
-      "instead of JavaScript, setting this to false will prevent the native module "
-      "from handling the events. Default: true")
-    static void SetEnableBackHandler(ReactInstanceSettings settings, Boolean value);
+      "By default `react-native-windows` will handle various back events and forward them to JavaScript. "
+      "Setting this to `BackNavigationHandlerKind.Native` prevents `react-native-windows` from handling "
+      "these events.  This will allow applications to handle back navigation in native code, but will prevent "
```
Comment (dannyvv) [inline vnext/Microsoft.ReactNative/QuirkSettings.idl:39]: sorry for another suggestion:
maybe:
prevents `react-native-windows` form handling these events, including forwarding to JavaScript.

**Review by dannyvv — APPROVED** [review]

File: vnext/Microsoft.ReactNative/QuirkSettings.idl (hunk 1 above)
Comment (dannyvv) [reply vnext/Microsoft.ReactNative/QuirkSettings.idl:39]: Note now it clicked what the original wording was. I got confused by the term 'native code' We were talking about 2 types of native code: a) Our native code in rnw that does the forwarding and b) customer native code.

File: vnext/Microsoft.ReactNative/QuirkSettings.idl [hunk 2]
```diff
@@ -28,10 +34,10 @@ namespace Microsoft.ReactNative
     static void SetAcceptSelfSigned(ReactInstanceSettings settings, Boolean value);
 
     DOC_STRING(
-      "Runtime setting that can be used to skip the default back handler logic.  "
-      "If you have an application that wants to handle back bahavior in native code "
-      "instead of JavaScript, setting this to false will prevent the native module "
-      "from handling the events. Default: true")
-    static void SetEnableBackHandler(ReactInstanceSettings settings, Boolean value);
+      "By default `react-native-windows` will handle various back events and forward them to JavaScript. "
+      "Setting this to `BackNavigationHandlerKind.Native` prevents `react-native-windows` from handling "
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/QuirkSettings.idl:38]:
```suggestion
      "Setting this to @BackNavigationHandlerKind.Native prevents `react-native-windows` from handling "
```

File: vnext/Microsoft.ReactNative/QuirkSettings.idl [hunk 3]
```diff
@@ -28,10 +34,10 @@ namespace Microsoft.ReactNative
...
+      "these events, including forwarding to JavaScript.  This will allow applications to handle back "
+      "navigation in native code, but will prevent the BackHandler native module from receiving events.")
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/QuirkSettings.idl:40]:
```suggestion
      "navigation in native code, but will prevent the `BackHandler` native module from receiving events.")
```

**Review by asklar — APPROVED** [review]
thanks a lot :)

File: vnext/Microsoft.ReactNative/ReactRootView.cpp [hunk 4]
```diff
@@ -363,8 +363,10 @@ void ReactRootView::AttachBackHandlers() noexcept {
   if (::Microsoft::ReactNative::IsXamlIsland())
     return;
 
-  if (!winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetEnableBackHandler(
-          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())))
+  if (winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetBackHandlerKind(
+          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties()) !=
+              winrt::Microsoft::ReactNative::BackNavigationHandlerKind
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactRootView.cpp:368]:
```suggestion
              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript))
```

File: vnext/Microsoft.ReactNative/ReactRootView.cpp [hunk 5]
```diff
@@ -363,8 +363,10 @@ void ReactRootView::AttachBackHandlers() noexcept {
...
+          : JavaScript))
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactRootView.cpp:369]:
```suggestion
              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript))
```

File: vnext/Microsoft.ReactNative/ReactRootView.cpp [hunk 6]
```diff
@@ -363,8 +363,9 @@ void ReactRootView::AttachBackHandlers() noexcept {
   if (::Microsoft::ReactNative::IsXamlIsland())
     return;
 
-  if (!winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetEnableBackHandler(
-          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())))
+  if (winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetBackHandlerKind(
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactRootView.cpp:366]:
```suggestion
  if (winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetBackHandlerKind(
          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())) !=
              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript)
```

File: vnext/Microsoft.ReactNative/ReactRootView.cpp [hunk 7]
```diff
@@ -363,8 +363,9 @@ void ReactRootView::AttachBackHandlers() noexcept {
...
+          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties()) !=
+              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript))
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactRootView.cpp:368]:
```suggestion
  if (winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetBackHandlerKind(
          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())) !=
              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript)
```

File: vnext/Microsoft.ReactNative/ReactRootView.cpp [hunk 8]
```diff
@@ -363,8 +363,9 @@ void ReactRootView::AttachBackHandlers() noexcept {
   if (::Microsoft::ReactNative::IsXamlIsland())
     return;
 
-  if (!winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetEnableBackHandler(
-          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())))
+  if (winrt::Microsoft::ReactNative::implementation::QuirkSettings::GetBackHandlerKind(
+          winrt::Microsoft::ReactNative::ReactPropertyBag(m_context->Properties())) !=
+              winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript)
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/ReactRootView.cpp:368]:
```suggestion
      winrt::Microsoft::ReactNative::BackNavigationHandlerKind::JavaScript)
```

File: vnext/Microsoft.ReactNative/QuirkSettings.cpp [hunk 9]
```diff
@@ -52,10 +54,10 @@ winrt::Microsoft::ReactNative::ReactPropertyId<bool> EnableBackHandlerProperty()
   ReactPropertyBag(settings.Properties()).Set(AcceptSelfSignedCertsProperty(), value);
 }
 
-/*static*/ void QuirkSettings::SetEnableBackHandler(
+/*static*/ void SetBackHandlerKind(
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/QuirkSettings.cpp:57]:
```suggestion
/*static*/ void QuirkSettings::SetBackHandlerKind(
```

**Pull request merged**